File: Tutorials/StylingExample/netcore/MainWindow.cs
using SpaceVIL;
using SpaceVIL.Common;
using SpaceVIL.Core;

namespace StylingExample
{
    public class MainWindow : ActiveWindow
    {
        // implement ActiveWindow class
        public override void InitWindow()
        {
            // add style for our Picture class to the current theme
            DefaultsService.GetDefaultTheme().AddDefaultCustomItemStyle(typeof(Picture), StyleFactory.GetPictureStyle());
            // slightly change style for WrapGrid
            StyleFactory.UpdateWrapGridStyle();

            // window attr
            SetParameters("MainWindow", "StylingExample", 800, 600);
            SetAntiAliasingQuality(MSAA.MSAA8x);

            // create WrapGrid as main layout
            WrapGrid layout = new WrapGrid(160, 120, Orientation.Horizontal);
            layout.SetStretch(true);

            // add WrapGrid to window
            AddItem(layout);

            // create and add 8 Picture items, the style for Picture class is automatically applied from the current theme
            layout.AddItems(
                    new Picture(DefaultsService.GetDefaultImage(EmbeddedImage.User, EmbeddedImageSize.Size64x64),
                            "User Image"),
                    new Picture(DefaultsService.GetDefaultImage(EmbeddedImage.Add, EmbeddedImageSize.Size64x64),
                            "Add Image"),
                    new Picture(DefaultsService.GetDefaultImage(EmbeddedImage.Eraser, EmbeddedImageSize.Size64x64),
                            "Eraser Image"),
                    new Picture(DefaultsService.GetDefaultImage(EmbeddedImage.Filter, EmbeddedImageSize.Size64x64),
                            "Filter Image"),
                    new Picture(DefaultsService.GetDefaultImage(EmbeddedImage.Gear, EmbeddedImageSize.Size64x64),
                            "Gear Image"),
                    new Picture(DefaultsService.GetDefaultImage(EmbeddedImage.Home, EmbeddedImageSize.Size64x64),
                            "Home Image"),
                    new Picture(DefaultsService.GetDefaultImage(EmbeddedImage.Refresh, EmbeddedImageSize.Size64x64),
                            "Refresh Image"),
                    new Picture(DefaultsService.GetDefaultImage(EmbeddedImage.Pencil, EmbeddedImageSize.Size64x64),
                            "Pencil Image")
                            );
        }
    }
}

File: Tutorials/StylingExample/jvm/src/main/java/StylingExample/MainWindow.java
package StylingExample;

import com.spvessel.spacevil.ActiveWindow;
import com.spvessel.spacevil.WrapGrid;
import com.spvessel.spacevil.Common.DefaultsService;
import com.spvessel.spacevil.Flags.EmbeddedImage;
import com.spvessel.spacevil.Flags.EmbeddedImageSize;
import com.spvessel.spacevil.Flags.MSAA;
import com.spvessel.spacevil.Flags.Orientation;

public class MainWindow extends ActiveWindow {
    // implement ActiveWindow class
    @Override
    public void initWindow() {
        // add style for our Picture class to the current theme
        DefaultsService.getDefaultTheme().addDefaultCustomItemStyle(Picture.class, StyleFactory.getPictureStyle());
        // slightly change style for WrapGrid
        StyleFactory.updateWrapGridStyle();

        // window attr
        setParameters("MainWindow", "StylingExample", 800, 600);
        setAntiAliasingQuality(MSAA.MSAA_8X);

        // create WrapGrid as main layout
        WrapGrid layout = new WrapGrid(160, 120, Orientation.HORIZONTAL);
        layout.setStretch(true);

        // add WrapGrid to window
        addItem(layout);

        // create and add 8 Picture items, the style for Picture class is automatically applied from the current theme
        layout.addItems(
                new Picture(DefaultsService.getDefaultImage(EmbeddedImage.USER, EmbeddedImageSize.SIZE_64X64),
                        "User Image"),
                new Picture(DefaultsService.getDefaultImage(EmbeddedImage.ADD, EmbeddedImageSize.SIZE_64X64),
                        "Add Image"),
                new Picture(DefaultsService.getDefaultImage(EmbeddedImage.ERASER, EmbeddedImageSize.SIZE_64X64),
                        "Eraser Image"),
                new Picture(DefaultsService.getDefaultImage(EmbeddedImage.FILTER, EmbeddedImageSize.SIZE_64X64),
                        "Filter Image"),
                new Picture(DefaultsService.getDefaultImage(EmbeddedImage.GEAR, EmbeddedImageSize.SIZE_64X64),
                        "Gear Image"),
                new Picture(DefaultsService.getDefaultImage(EmbeddedImage.HOME, EmbeddedImageSize.SIZE_64X64),
                        "Home Image"),
                new Picture(DefaultsService.getDefaultImage(EmbeddedImage.REFRESH, EmbeddedImageSize.SIZE_64X64),
                        "Refresh Image"),
                new Picture(DefaultsService.getDefaultImage(EmbeddedImage.PENCIL, EmbeddedImageSize.SIZE_64X64),
                        "Pencil Image")
                        );
    }
}